File: src/main/java/com/morak/performancetracker/aop/persistence/ConnectionAdvisor.java
package com.morak.performancetracker.aop.persistence;

import org.aopalliance.aop.Advice;
import org.springframework.aop.Pointcut;
import org.springframework.aop.PointcutAdvisor;

public class ConnectionAdvisor implements PointcutAdvisor {

    private final Pointcut pointcut;
    private final Advice advice;

    public ConnectionAdvisor(Pointcut pointcut, Advice advice) {
        this.pointcut = pointcut;
        this.advice = advice;
    }

    @Override
    public Pointcut getPointcut() {
        return this.pointcut;
    }

    @Override
    public Advice getAdvice() {
        return this.advice;
    }

    @Override
    public boolean isPerInstance() {
        return true;
    }
}

File: src/main/kotlin/com/morak/performancetracker/aop/persistence/ConnectionAdvisor.kt
package com.morak.performancetracker.aop.persistence

import org.aopalliance.aop.Advice
import org.springframework.aop.Pointcut
import org.springframework.aop.PointcutAdvisor

class ConnectionAdvisor(private val pointcut: Pointcut, private val advice: Advice) : PointcutAdvisor {
    override fun getPointcut(): Pointcut {
        return pointcut
    }

    override fun getAdvice(): Advice {
        return advice
    }

    override fun isPerInstance(): Boolean {
        return true
    }
}
